internal/controller/postgrescluster/postgres.go

```diff
@@ -339,13 +339,13 @@ func (r *Reconciler) reconcilePostgresUserSecrets(
 		if n := len(cluster.Name); n > 63 {
 			allErrors = append(allErrors,
 				field.Invalid(path, cluster.Name,
-					fmt.Sprintf("should be at most %d chars long", 63)))
+					fmt.Sprintf("should be at most %d chars long (a RFC 1123 label)", 63)))
 		}
 		// See v1beta1.PostgresRoleSpec validation markers.
```

Comment:
Is there a URL we could put here for v1beta1.PostgresRoleSpec? Was not able to find it during a quick try to do so -- not sure if it was a GitHub search limitation.

Reply:
I'd be inclined to point at Kubernetes naming conventions (i.e. https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names). @cbandy what are your thoughts?

Reply:
These messages only occur when the _cluster_ name is not a valid user name. These lines do not affect the `user.name` field of the spec where we are using Kubernetes' OpenAPI validation.

We call out this limitation in the field description. The [API reference](https://access.crunchydata.com/documentation/postgres-operator/v5/references/crd/#postgresclusterspecusersindex) in the docs might be somewhere to link.

https://github.com/CrunchyData/postgres-operator/blob/99fc484e6a459c07ebb26c60c27f89b2f702b1df/pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgres_types.go#L21-L23
https://github.com/CrunchyData/postgres-operator/blob/99fc484e6a459c07ebb26c60c27f89b2f702b1df/config/crd/bases/postgres-operator.crunchydata.com_postgresclusters.yaml#L7562-L7569


As I recall, these messages are intentionally similar to the ones that Kubernetes emits through OpenAPI validation.

Reply:
Unfortunately, `kubectl explain` shows only type, optional/required, and the description.

```console
$ kubectl explain postgrescluster.spec.users
…
FIELDS:
   …

   name <string> -required-
     The name of this PostgreSQL user. The value may contain only lowercase
     letters, numbers, and hyphen so that it fits into Kubernetes metadata.

   …
```